example/src/Shared/Contacts.tsx

```diff
@@ -56,46 +56,45 @@ const CONTACTS: Item[] = [
   { name: 'Vincent Sandoval', number: 2606111495 },
 ];
 
-class ContactItem extends React.PureComponent<{
-  item: { name: string; number: number };
-}> {
-  render() {
-    const { item } = this.props;
-
-    return (
-      <View style={styles.item}>
-        <View style={styles.avatar}>
-          <Text style={styles.letter}>
-            {item.name.slice(0, 1).toUpperCase()}
-          </Text>
-        </View>
-        <View style={styles.details}>
-          <Text style={styles.name}>{item.name}</Text>
-          <Text style={styles.number}>{item.number}</Text>
-        </View>
+const ContactItem = ({ item: { name, number } }: { item: Item }) => {
+  return (
+    <View style={styles.item}>
+      <View style={styles.avatar}>
+        <Text style={styles.letter}>{name.slice(0, 1).toUpperCase()}</Text>
+      </View>
+      <View style={styles.details}>
+        <Text style={styles.name}>{name}</Text>
+        <Text style={styles.number}>{number}</Text>
       </View>
-    );
-  }
-}
+    </View>
+  );
+};
 
-export default class Contacts extends React.Component {
-  private renderItem = ({ item }: { item: Item }) => (
-    <ContactItem item={item} />
+const Contacts = () => {
+  const renderItem = React.useCallback((item: { item: Item }) => {
```

Comment:
You should use types provided by FlatList to get the item type. The same applies to keyExtractor